public/service/config_interpolated_string.go

```diff
@@ -87,3 +101,36 @@ func (p *ParsedConfig) FieldInterpolatedStringMap(path ...string) (map[string]*I
 	}
 	return sMap, nil
 }
+
+// FieldInterpolatedStringList accesses a field that is a list of interpolated
+// string values from the parsed config.
+//
+// Returns an error if the field is not found, or is not an list of interpolated
+// strings.
+func (p *ParsedConfig) FieldInterpolatedStringList(path ...string) ([]*InterpolatedString, error) {
+	v, exists := p.field(path...)
+	if !exists {
+		return nil, fmt.Errorf("field '%v' was not found in the config", p.fullDotPath(path...))
+	}
+
+	raw, ok := v.([]any)
+	if !ok {
+		return nil, fmt.Errorf("expected field '%v' to be a string list, got %T", p.fullDotPath(path...), v)
+	}
+
+	values := make([]*InterpolatedString, len(raw))
+	for i, rawValue := range raw {
+		var parsed string
+		var ok bool
+		if parsed, ok = rawValue.(string); !ok {
+			return nil, fmt.Errorf("expected field '%v' to be a string list, found an element of type %T", p.fullDotPath(path...), rawValue)
+		}
+
+		e, err := p.mgr.BloblEnvironment().NewField(parsed)
+		if err != nil {
+			return nil, fmt.Errorf("%w '%v': %v", errInvalidInterpolation, strings.Join(path, "."), err)
+		}
+		values[i] = &InterpolatedString{expr: e}
```

Comment:
Small nit: using append with make(T, 0, len) will avoid initialising the memory with default values